### ISSUE TITLE
Support for WAVE parameters

### ISSUE THREAD

**Issue by paulray** [nanograv/PINT#381]
PINT does not support WAVE parameters:

```
WARNING: Unrecognized parfile line 'WAVEEPOCH 58074' [pint.models.timing_model]
WARNING: Unrecognized parfile line 'WAVE_OM 0.18150803212366 0' [pint.models.timing_model]
WARNING: Unrecognized parfile line 'WAVE1 -2.3580237569833 0.61257000451779' [pint.models.timing_model]
WARNING: Unrecognized parfile line 'WAVE2 0.17435474955813 -0.096065138705814' [pint.models.timing_model]
WARNING: Unrecognized parfile line 'WAVE3 -0.020362229321081 0.019154948126028' [pint.models.timing_model]
WARNING: Unrecognized parfile line 'WAVE4 0.0019083164479827 -0.0031306122851833' [pint.models.timing_model]
WARNING: Unrecognized parfile line 'WAVE5 -6.5280719905701e-05 0.00028346991238931' [pint.models.timing_model]
```

This should be added...

**Comment by paulray** [comment]
@luojing1211 I'm thinking of adding this model. Is there a HOWTO somewhere that give a step-by-step guide to adding a model to PINT? I know the structure is described in the paper, but a simple guide on the wiki would be great for people who want to contribute new models.

There is a tiny bit here: https://github.com/nanograv/PINT/wiki/Code-Architecture but it really needs more detail on what a programmer needs to define for a new model (e.g. derivatives and such).

**Comment by luojing1211** [comment]
Hi Paul,

I was thinking to add this part. So how about give an example code, and we
provide the guide from the example code.

-Jing Luo

**Comment by paulray** [comment]
Is this working now? Can we close?

**Comment by aarchiba** [comment]
There is `test_wave.py`; is WAVE now sufficiently tested to close this bug?

**Comment by matteobachetti** [comment]
@aarchiba I tried to use extensively the WAVE parameters. If I'm not mistaken, they work well if you just take them at face value, but _fitting_ them leads to inconsistent results

**Comment by abhisrkckl** [comment]
Can this be closed?

**Comment by kerrm** [comment]
I think the status is still the same -- WAVE parameters work fine for an
input model, but you can't fit them.

**Comment by paulray** [comment]
Is the problem that there are 2 parameters for each WAVE?

**Comment by kerrm** [comment]
I think that's the crux.  It's also a "prefix parameter", which makes
things a little more complicated, but the pair is the big problem.

I'd say the best approach would be to map them to internal parameters
WAVE1_COS, WAVE1_SIN, etc. then translate back to the WAVE model for
output.  There's a further difference that the "WAVE" format doesn't
support uncertainty reporting.  But maybe that could be encapsulated with
the future covariance reporting Anne would like to do.

**Comment by abhisrkckl** [comment]
We have decided not to fix Wave. 
An alternative is available in the WaveX model -- #1609
